static/app/views/releases/detail/withReleaseRepos.tsx

```diff
@@ -36,198 +36,188 @@ type State = {
-            } = this.props;
-
-            if (isLoading || repositoriesLoading) {
-              return <LoadingIndicator />;
-            }
-
-            const noRepositoryOrgRelatedFound = !repositories?.length;
-
-            if (noRepositoryOrgRelatedFound) {
-              const {orgId} = params;
-              return (
-                <Body>
-                  <Main fullWidth>
-                    <Panel dashedBorder>
-                      <EmptyMessage
-                        icon={<IconCommit size="xl" />}
-                        title={t('Releases are better with commit data!')}
-                        description={t(
-                          'Connect a repository to see commit info, files changed, and authors involved in future releases.'
-                        )}
-                        action={
-                          <Button priority="primary" to={`/settings/${orgId}/repos/`}>
-                            {t('Connect a repository')}
-                          </Button>
-                        }
-                      />
-                    </Panel>
-                  </Main>
-                </Body>
-              );
-            }
-
-            const noReleaseReposFound = !releaseRepos.length;
-
-            if (noReleaseReposFound) {
-              return (
-                <Body>
-                  <Main fullWidth>
-                    <Panel dashedBorder>
-                      <EmptyMessage
-                        icon={<IconCommit size="xl" />}
-                        title={t('Releases are better with commit data!')}
-                        description={t(
-                          'No commits associated with this release have been found.'
-                        )}
-                      />
-                    </Panel>
-                  </Main>
-                </Body>
-              );
-            }
-
-            if (activeReleaseRepo === undefined) {
-              return <LoadingIndicator />;
-            }
-
-            const {release} = params;
-            const orgSlug = organization.slug;
-
-            return (
-              <WrappedComponent
-                {...(this.props as P)} // this is just to satisfy the compiler
-                orgSlug={orgSlug}
-                projectSlug={this.context.project.slug}
-                release={release}
-                router={router}
-                location={location}
-                releaseRepos={releaseRepos}
-                activeReleaseRepo={activeReleaseRepo}
-              />
-            );
-          }
-        }
-      )
-    )
-  );
+) {
+  class WithReleaseRepos extends React.Component<P & HoCsProps, State> {
+    static displayName = `withReleaseRepos(${getDisplayName(WrappedComponent)})`;
+
+    state: State = {
+      releaseRepos: [],
+      isLoading: true,
+    };
+
+    componentDidMount() {
+      this.fetchReleaseRepos();
+    }
+
+    UNSAFE_componentWillReceiveProps(nextProps: P & HoCsProps) {
```

Comment:
Any chance you can clean this up while you're in here?

Reply:
I can come back to it and fix this up afterwards. I'd like to finish off with the createReactClass replacements as there are only a few left.